extensions/panache/rest-data-panache/runtime/src/main/java/io/quarkus/rest/data/panache/ResourceProperties.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.rest.data.panache;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RUNTIME)
+@Target({ TYPE })
+public @interface ResourceProperties {
+
+    /**
+     * Generate operations that support HAL content type.
+     * HAL methods are generated in addition to the standard methods. They accept the same parameters but return a content of
+     * `application/hal+json` type.
+     * The operations that support HAL responses are `get`, `list`, `add` and `update`.
+     *
+     * Default: false.
+     */
+    boolean hal() default false;
+
+    /**
+     * URL path segment that should be used to access the resources.
+     *
+     * Default: hyphenated resource name without a suffix. Ignored suffixes are `Controller` and `Resource`.
+     */
+    String path() default "";
```

Comment:
Not that I think this will ever be an issue, but by using `""` as the default value, you are effectively preventing users from making an panache rest data resource without a path. Now that's probably A good thing, but I am just bringing it up in case you hadn't considered it. In any case it would probably be safer to use some nonsensical default valur

Reply:
Hmm, we could maybe change this later too. Adding a nonsense value and based on that not add an annotation at all. I need to add some default value because otherwise user will always have to specify a path when using this annotation.

Reply:
Yeah, I understand the need for a default value, just saying that `""` is not the proper one :)

Reply:
Shall we change it separately after the release? I don't want to hold this PR for too long.

Reply:
Sure yeah